HopsanGUI/Utilities/GUIUtilities.cpp

```diff
@@ -519,7 +519,14 @@ void reduceVectorSize(QVector<double> &vector, int newSize)
 
     for(int i=0; i<newSize; ++i)
     {
-        tempVector.append(vector.at(oldSize/newSize*i));
+        double pos = double(oldSize)/double(newSize)*double(i);
+        int prev = int(floor(pos)); //Previous index in original vector
+        int next = int(ceil(pos));  //Next index in original vector
+        double x = 0;
+        if(next != prev) {
+            x = (pos-prev)/(next-prev);  //Fraction
+        }
+        tempVector.append((1.0-x)*vector.at(prev) + x*vector.at(next));
```

Comment:
Since newSize is known, the vector memory should be reserved before the loop

I know that this function is meant to reduce vector, but what if newSize is >=oldSize, then I think that vector.at() will throw
Maybe protect against that with in initial if check, and just do no do anything if newSize is not smaller

Reply:
Ok, I added preallocation and made some modifications to prevent going out of bounds. It should now work even if newSize is larger.

Reply:
I also renamed the function to `resampleVector`, to make clear that it works for both upsampling and downsampling.